lib/MediaConnection/RTCPeerMediaConnection.js

```diff
@@ -0,0 +1,171 @@
+    {
+        const description = new RTCSessionDescription({ type: type, sdp: sdp });
+
+        return this.rtcPeerConnection.setRemoteDescription(description);
+    }
+
+    addTrack(track, stream) {
+        this.rtcPeerConnection.addTrack(track, stream);
+    }
+
+    getLocalSdp(constraints, type)
+    {
+        if (this.iceGatheringState === 'complete' && (!constraints || !constraints.iceRestart))
+        {
+            const e = {originator: 'local', type: type, sdp: this.getLocalSdp().sdp};
+
+            return Promise.resolve(e);
+        }
+
+        return new Promise(resolve =>
+        {
+            let finished = false;
+            let iceCandidateListener;
+            let iceGatheringStateListener;
+
+            const ready = () =>
+            {
+                this.rtcPeerConnection.removeEventListener('icecandidate', iceCandidateListener);
+                this.rtcPeerConnection.removeEventListener('icegatheringstatechange', iceGatheringStateListener);
+
+                finished = true;
+
+                const e = {originator: 'local', type: type, sdp: this._getLocalSdp().sdp};
+                resolve(e)
+            };
+
+            this.rtcPeerConnection.addEventListener('icecandidate', iceCandidateListener = (event) =>
+            {
+                const candidate = event.candidate;
+
+                if (candidate)
+                {
+                    this.emit('icecandidate',
+                        {
+                        candidate: event.candidate,
+                        ready: ready
+                    });
+                } else if (!finished)
+                {
+                    ready();
+                }
+            });
+
+            this.rtcPeerConnection.addEventListener('icegatheringstatechange', iceGatheringStateListener = () =>
+            {
+                if ((this.iceGatheringState === 'complete') && !finished) {
+                    ready();
+                }
+            });
+        })
+    }
+
+    sendInbandDTMF(tones, duration, interToneGap)
+    {
+        const getDTMFSender = () =>
+        {
+            const sender = this.rtcPeerConnection.getSenders().find((rtpSender) =>
+            {
+                return rtpSender.track && rtpSender.track.kind === 'audio';
+            });
+
+            if (!(sender && sender.dtmf))
+            {
+                return;
+            }
+
+            return sender.dtmf;
+        }
+
+        getDTMFSender().insertDTMF(tones, duration, interToneGap)
+    }
+
+    close()
+    {
+        // TODO: close streams
+        this.rtcPeerConnection.close()
+    }
+
+    /**
+     * RtcPeer specific methods for the opinionated default implementation
+     */
+
+    // Used by exposed peer-connection (tryit-jssip)
+    addEventListener(onEvent, callback)
+    {
+        this.rtcPeerConnection.addEventListener(onEvent, callback)
+    }
+
+    // TODO: used by un/mute functions. How to handle this?
+    getSenders()
```

Comment:
In order not to expose `RtpSenders` we can just create the method `getLocalTracks()`, and for consistency `getRemoteTracks()`